tests/test_activate.py

```diff
@@ -1,74 +1,194 @@
+        #    shell_suffix=".ps",
+        #    printps1='echo $PS1',
+        #    printdefaultenv='echo $CONDA_DEFAULT_ENV',
+        #    printpath="echo %PATH%",
+        #    raw_ps=os.getenv("PROMPT", ""),
+        #    exe="powershell.exe",
+        #    path_from=path_identity,
+        #    path_to=path_identity,
+        #    slash_convert = (u"/", u"\\"),
+        #),
+        "cmd.exe": dict(
+            echo="echo",
+            ps_var="PROMPT",
+            var_format="%{}%",
+            binpath="\\Scripts\\",  # mind the trailing slash.
+            source_setup="call",
+            test_echo_extra="",
+            nul='1>NUL 2>&1',
+            set_var='set ',
+            shell_suffix=".bat",
+            printps1="echo %PROMPT%",
+            printdefaultenv='IF NOT "%CONDA_DEFAULT_ENV%" == "" (\n'
+                            'echo %CONDA_DEFAULT_ENV% ) ELSE (\n'
+                            'echo()', # parens mismatched intentionally.  See http://stackoverflow.com/questions/20691060/how-do-i-echo-a-blank-empty-line-to-the-console-from-a-windows-batch-file
+            printpath="echo %PATH%",
+            raw_ps=os.getenv("PROMPT", ""),
+            exe="cmd.exe",
+            shell_args="/d /c",
+            path_from=path_identity,
+            path_to=path_identity,
+            slash_convert = (u"/", u"\\"),
+        ),
+        "cygwin": dict(unix_shell_base, exe="c:\\cygwin\\bin\\bash", path_from=cygwin_path_to_win, path_to=win_path_to_cygwin),
+        "msys": dict(unix_shell_base, exe="C:\\msys\\1.0\\bin\\bash", path_from=unix_path_to_win, path_to=win_path_to_unix),
+        "msys2": dict(unix_shell_base, exe="C:\\msys\\2.0\\usr\\bin\\bash", path_from=unix_path_to_win, path_to=win_path_to_unix),
+    }
 
-# Only run these tests for commands that are installed.
+else:
+    shells = {
+        "bash": dict(unix_shell_base, exe="bash"),
+        "zsh": dict(unix_shell_base, exe="zsh"),
+    }
+
+def run_in(command, shell):
+    if shell == 'cmd.exe':
+        cmd_script = tempfile.NamedTemporaryFile(suffix='.bat', mode='wt', delete=False)
+        cmd_script.write(command)
+        cmd_script.close()
+        cmd_bits = [shells[shell]["exe"]] + shells[shell]["shell_args"].split(" ") + [cmd_script.name]
+        p = subprocess.Popen(cmd_bits, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        os.unlink(cmd_script.name)
+    elif shell == 'powershell':
+        raise NotImplementedError
+    else:
+        cmd_bits = [shells[shell]["exe"]] + shells[shell]["shell_args"].split(" ") + [translate_stream(command, shells[shell]["path_to"])]
+        print(cmd_bits)
+        p = subprocess.Popen(cmd_bits, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+    output_translator = shells[shell]["path_from"]
+    streams = [translate_stream(stream.strip().decode('utf-8').replace('\r\n', '\n').replace('\\\\', '\\'), output_translator)
```

Comment:
Same comment as [before](https://github.com/conda/conda/pull/1727/files#r51996742): Why do you have to `replace('\\\\', '\\')` here? What produces these double backslashes?